File: WAS-backend/Models/OrdreProduction.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WAS_backend.Models;

public class OrdreProduction
{
    public int       Id                    { get; set; }
    public DateTime? Horodatage            { get; set; }
    public int       Statut                { get; set; } = 0;

    // ✅ CORRIGÉ — non nullable
    public string    Numero                { get; set; } = string.Empty;

    [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
    public string?   NumeroAuto            { get; set; }
    public string?   Description           { get; set; }
    public string?   DescriptionRecherche  { get; set; }
    public string?   Description2          { get; set; }
    public DateTime? DateCreation          { get; set; }
    public DateTime? DateDerniereModif     { get; set; }
    public int?      TypeSource            { get; set; }
    public string?   NumeroSource          { get; set; }
    public string?   NumeroGamme           { get; set; }
    public string?   CodeVariante          { get; set; }
    public string?   GroupeStockInventaire { get; set; }
    public string?   GroupeComptaProd      { get; set; }
    public string?   GroupeComptaAffaires  { get; set; }
    public TimeSpan? HeureDebut            { get; set; }
    public DateTime? DateDebut             { get; set; }
    public TimeSpan? HeureFin              { get; set; }
    public DateTime? DateFin               { get; set; }
    public DateTime? DatePrevue            { get; set; }
    public DateTime? DateFinReelle         { get; set; }
    public bool      Bloque                { get; set; } = false;
    public string?   CodeDimension1        { get; set; }
    public string?   CodeDimension2        { get; set; }
    public string?   CodeSite              { get; set; }
    public string?   CodeEmplacement       { get; set; }
    public string?   NumeroReplanRef       { get; set; }
    public int?      StatutReplanRef       { get; set; }
    public int?      CodeNiveauBas         { get; set; }
    public decimal?  Quantite              { get; set; }
    public decimal?  CoutUnitaire          { get; set; }
    public decimal?  MontantCout           { get; set; }
    public string?   SerieNumero           { get; set; }
    public string?   NumeroOrdrePrevu      { get; set; }
    public string?   NumeroOrdreFerme      { get; set; }
    public string?   NumeroOrdreSimule     { get; set; }
    public DateTime? DateHeureDebut        { get; set; }
    public DateTime? DateHeureFin          { get; set; }
    public int?      IdEnsembleDimension   { get; set; }
    public string?   OperateurAssigne      { get; set; }

    // Navigation
    public ICollection<LigneOrdreProduction> Lignes     { get; set; } = new List<LigneOrdreProduction>();
    public ICollection<ComposantOrdre>       Composants { get; set; } = new List<ComposantOrdre>();
    public ICollection<OperationGamme>       Operations { get; set; } = new List<OperationGamme>();
}

File: was-frontend/src/features/ordres-production/models/OrdreProduction.ts
export interface OrdreProduction {
  id: number
  horodatage?: string
  statut: number
  numero: string
  numeroAuto?: string        // ← AJOUT
  description?: string
  descriptionRecherche?: string
  description2?: string
  dateCreation?: string
  dateDerniereModif?: string
  typeSource?: number
  numeroSource?: string
  numeroGamme?: string
  codeVariante?: string
  dateDebut?: string
  dateFin?: string
  datePrevue?: string
  dateFinReelle?: string
  bloque?: boolean
  codeSite?: string
  codeEmplacement?: string
  quantite?: number
  coutUnitaire?: number
  montantCout?: number
  operateurAssigne?: string
  dateHeureDebut?: string
  dateHeureFin?: string
}

export interface LigneOrdreProduction {
  id: number
  statut: number
  numeroOrdreProduction: string
  numeroLigne?: number
  referenceArticle?: string
  description?: string
  quantite?: number
  quantiteTerminee?: number
  quantiteRestante?: number
  tauxRebut?: number
  dateDebut?: string
  dateFin?: string
  uniteMesure?: string
  numeroBOM?: string
}

export interface ComposantOrdre {
  id: number
  statut: number
  numeroOrdreProduction: string
  referenceArticle?: string
  description?: string
  uniteMesure?: string
  quantite?: number
  quantiteAttendue?: number
  quantiteRestante?: number
  tauxRebut?: number
  dateBesoin?: string
}

export interface OperationGamme {
  id: number
  statut: number
  numeroOrdreProduction: string
  numeroOperation?: string
  description?: string
  numeroCentreTravail?: string
  groupeCentreTravail?: string
  tempsReglage?: number
  tempsExecution?: number
  dateDebut?: string
  dateFin?: string
  statutGamme?: number
}

export interface OrdreProductionDetailDTO {
  ordre: OrdreProduction
  lignes: LigneOrdreProduction[]
  composants: ComposantOrdre[]
  operations: OperationGamme[]
}

export const STATUT_LABELS: Record<number, string> = {
  0: 'Simulé',
  1: 'Planifié',
  2: 'En cours',
  3: 'Terminé',
}

export const STATUT_COLORS: Record<number, string> = {
  0: '#6B7280',
  1: '#3B82F6',
  2: '#F59E0B',
  3: '#10B981',
}

export const STATUT_BG: Record<number, string> = {
  0: '#F3F4F6',
  1: '#EFF6FF',
  2: '#FFFBEB',
  3: '#F0FDF4',
}